packages/mattermost-redux/src/selectors/entities/channel_categories.ts

```diff
@@ -472,26 +471,20 @@ export function makeSortChannels() {
 export function makeGetChannelIdsForCategory() {
     const getChannels = makeGetChannelsForIds();
     const filterAndSortChannelsForCategory = makeFilterAndSortChannelsForCategory();
-    let prevChannels: Channel[] = [];
-    let prevFilteredChannelIds: string[] = [];
+
+    let lastChannelIds: string[] = [];
 
     return (state: GlobalState, category: ChannelCategory) => {
         const channels = getChannels(state, category.channel_ids);
 
-        if (channels.length === 0 && prevChannels.length === 0) {
-            return prevFilteredChannelIds;
-        }
+        const filteredChannelIds = filterAndSortChannelsForCategory(state, channels, category).map((channel) => channel.id);
 
-        // getChannels will almost never return a memoized result.
-        // Assumption here is that the added comparison computation is more beneficial
-        // than the cost of returning a non memoized result.
-        if (deepEqual(channels, prevChannels)) {
-            return prevFilteredChannelIds;
+        if (shallowEquals(filteredChannelIds, lastChannelIds)) {
```

Comment:
This was the big change for this one since we're checking equality of the filtered IDs instead of the prefiltered channels.